analysis/analysis_utils.py

```diff
@@ -238,53 +238,60 @@ def save_results(self, res_filepath, out_suffix, keep):
-                The WCS from a fits file, as loaded in with astropy.wcs.WCS()
-            angle_to_ecliptic : [NEEDS DOC STRING]
-        OUTPUT-
-            eclip_angle : float
-                The angle to the ecliptic for the fits file corresponding to
-                test_wcs. NOTE- May be off by +/- PI (180 degrees)
+        Projects an unit-vector parallel with the ecliptic onto the image
+        and calculates the angle of the projected unit-vector in the pixel
+        space.
+
+        Parameters
+        ----------
+        wcs : `astropy.wcs.WCS`
+            World Coordinate System object.
+        angle_to_ecliptic : `float`, optional
+            Angle to ecliptic, default: 0.0 
+
+        Returns
+        -------
+        angle : `float`
+            Angle the projected unit-vector parallel to the ecliptic 
+            closes with the image axes. Used to transform the specified
+            search angles, with respect to the ecliptic, to search angles
+            within the image.
+
+        Note
+        ----
+        It is not neccessary to calculate this angle for each image in an
+        image set if they have all been warped to a common WCS.
+
+        See Also
+        --------
+        run_search.do_gpu_search
         """
-        pixel_coords = [[], []]
-        pixel_start = [[1000, 2000]]
-        angle = float(angle_to_ecliptic)
-        vel_array = np.array([[6.0 * np.cos(angle), 6.0 * np.sin(angle)]])
-        time_array = [0.0, 1.0, 2.0]
-        vel_par_arr = vel_array[:, 0]
-        vel_perp_arr = vel_array[:, 1]
-
-        if type(vel_par_arr) is not np.ndarray:
-            vel_par_arr = [vel_par_arr]
-        if type(vel_perp_arr) is not np.ndarray:
-            vel_perp_arr = [vel_perp_arr]
-        for start_loc, vel_par, vel_perp in zip(pixel_start, vel_par_arr, vel_perp_arr):
-
-            start_coord = astroCoords.SkyCoord.from_pixel(start_loc[0], start_loc[1], test_wcs)
-            eclip_coord = start_coord.geocentrictrueecliptic
-            eclip_l = []
-            eclip_b = []
-            for time_step in time_array:
-                eclip_l.append(eclip_coord.lon + vel_par * time_step * u.arcsec)
-                eclip_b.append(eclip_coord.lat + vel_perp * time_step * u.arcsec)
-            eclip_vector = astroCoords.SkyCoord(eclip_l, eclip_b, frame="geocentrictrueecliptic")
-            pixel_coords_set = astroCoords.SkyCoord.to_pixel(eclip_vector, test_wcs)
-            pixel_coords[0].append(pixel_coords_set[0])
-            pixel_coords[1].append(pixel_coords_set[1])
-
-        pixel_coords = np.array(pixel_coords)
-        x_dist = pixel_coords[0, 0, -1] - pixel_coords[0, 0, 0]
-        y_dist = pixel_coords[1, 0, -1] - pixel_coords[1, 0, 0]
-        eclip_angle = np.arctan2(y_dist, x_dist)
-
-        return eclip_angle
+        # pick a starting pixel approximately near the center of the image
+        # convert it to ecliptic coordinates
+        start_pixel = np.array([1000, 2000])
```

Comment:
I know this is copied over from the original code, but I think this could be too brittle. It is effectively hardcoding the center of the image. Would it be better to automatically calculate the center of the image?

Reply:
Fixed, with a note that the actual center of the image is not exactly `(1000, 2000)` because image dimension for DEEP are `(1048, 2048)` or similar power of 2. I tested, forcing them to be exactly 100 and 200 against previous code and it produced the same values. Not sure how this affects comparing results, since these values seem arbitrary.